symengine/polynomial.cpp

```diff
@@ -22,9 +22,8 @@ UnivariateIntPolynomial::UnivariateIntPolynomial(const RCP<const Symbol> &var, c
 
 bool UnivariateIntPolynomial::is_canonical(const unsigned int &degree_, const map_uint_mpz& dict) const {
     if (var_->get_name() == "")
-        if (!(dict.empty() or dict.size() == 1))
+        if (!((dict.empty() or dict.size() == 1) and (dict.empty() or dict.begin()->first == 0)))
```

Comment:
`dict.empty() or (dict.size() == 1 and dict.begin()->first == 0)` is simpler than above.